src/sass/atoms/_typography.scss

```diff
@@ -9,56 +9,35 @@ $typographySizes: (
     zeta    : 0.8em
 ) !default;
 
-@include createMultipleSizeClasses('f', $typographySizes, ('font-size'));
-
-.f-normal {
-    font-style: normal;
-}
-
-.f-bold {
-    font-weight: bold;
-}
-
-.f-italic {
-    font-style:italic;
-}
-
-.f-w-normal {
-    font-weight: normal;
-}
-
-.f-w-700 {
-    font-weight: 700;
-}
-
-.f-uppercase {
-    text-transform: uppercase;
-}
-
-.f-line-through {
-    text-decoration: line-through;
-}
-
-.f-no-decoration {
-    text-decoration: none;
-}
-
-.f-underline {
-    text-decoration: underline;
-}
-
-.f-right {
-    text-align: right;
-}
-
-.f-left {
-    text-align: left;
-}
-
-.f-center {
-    text-align: center;
-}
-
-.f-no-wrap {
-    white-space: nowrap;
-}
+$typographyClasses: (
+    f-italic: ( font-style: italic ),
+    f-normal: ( font-style: normal ),
+    f-oblique: ( font-style: oblique ),
+    f-w-normal: ( font-weight: normal ),
+    f-w-bold: ( font-weight: bold ),
+    f-w-700: ( font-weight: 700 ),
+    f-uppercase: ( text-transform: uppercase ),
+    f-line-through: ( text-decoration: line-through ),
+    f-no-decoration: ( text-decoration: none ),
+    f-underline: ( text-decoration: underline ),
+    f-left: ( text-align: left ),
+    f-center: ( text-align: center ),
+    f-right: ( text-align: right ),
+    f-no-wrap: ( white-space: nowrap )
+);
+
+@function mergeTypographySizesAndClasses($prefix, $definitions, $mapToMergeWith) {
```

Comment:
How about changing this to more abstract function? Change $mapToMergeWith parameter into $properties, like ('font-size') ? Then move it into functions and here just make merge  $typographyAndFontSizeClasses: map-merge($typographyClasses, $fontSizeClasses);